guide/creating-your-bot/parsing-mention-arguments.md

```diff
@@ -0,0 +1,219 @@
+
+That means when you receive a message from the Discord API and it contains mentions the message's content will contain that special syntax.  
+If you send
+
+<discord-messages>
+	<discord-message author="User" avatar="djs">
+		I think we should add @GoodPerson to the @Mod role.
+	</discord-message>
+</discord-messages>
+
+then the `message.content` for that message will look something like this
+
+<!-- eslint-skip -->
+```js
+'I think we should add <@86890631690977280> to the <@&134362454976102401> role.'
+```
+
+### Implementation
+
+So, how do you actually use this new information for your bot?  
+Most of your code will not change, however instead of using `message.mentions` to find the mentioned users you will have to do it manually.  
+This may sound scary at first, but once you see the code you will see it is actually pretty simple.
+
+Say you already have a simple command handler like this:
+
+```js
+client.on('message', message => {
+	// Don not go any further if there is no command prefix.
+	if (!message.content.startsWith(config.prefix)) return;
+
+	// Remove the prefix from the message.
+	const withoutPrefix = message.content.slice(config.prefix.length);
+	// Split the message by spaces
+	const split = withoutPrefix.split(/ +/);
+	// The first element is the command, the rest are the arguments.
+	const command = split[0];
+	const args = split.slice(1);
+
+	// ....Some commands here....
+});
+```
+
+Now you can easily test the waters by upgrading the avatar command from last time.
+This is what we have so far. It is pretty simple, it will show the avatar of who used the command.
+
+```js
+if (command === 'avatar') {
+	// What to put here to use the mention?
+	const user = message.author;
+
+	return message.channel.send(`${user.username}'s avatar: ${user.displayAvatarURL}`);
+}
+```
+
+But how do you actually get the correct user now? Well, this requires a few simple steps.  
+Putting it into a function will make it easily reusable. We will use the name `getUserFromMention` here.
+
+```js
+function getUserFromMention(mention) {
+	// Check if `mention` contains a string.
+	// If not, just return undefined to signal there is no user to be found.
+	if (!mention) return undefined;
+
+	// A user mention starts with <@ as well as a > at the end,
+	// so check for those.
+	if (mention.startsWith('<@') && mention.endsWith('>')) {
+		// Now remove the <@ and >, they are not necessary.
+		mention = mention.slice(2, -1);
+
+		// If the user has a nickname they will also have a ! in their mention.
+		// Remove that as well.
+		if (mention.startsWith('!')) {
+			mention = mention.slice(1);
+		}
+
+		// Now all that is left should be the ID,
+		// which can be used to get the user object.
+		const user = client.users.get(mention);
+		// The ID might be invalid, so check for that as well.
+		if (!user) {
+			// Just return undefined to indicate nothing was found.
+			return undefined;
+		}
+		else {
```

Comment:
No need for the `else`. I think it can be simplified to just `if (user) return user;` and then maybe omit the `return undefined` entirely and let it hit the last line in the function.

Or maybe even just `return client.users.get(mention);`, because `.get()` returns undefined if nothing was found. In this case, if that were done, then a comment mentioning that would be okay, I think.